electron/src/trusted-main-electron-process/SecureDiceKeyStoreApi.ts

```diff
@@ -1,12 +1,111 @@
 import * as IpcApiFactory from "./IpcApiFactory"
 import * as keytar from 'keytar';
+import * as crypto from 'crypto';
+import * as fs from 'fs';
+import {promises as fsp} from "fs"
+import * as path from 'path';
+import {app} from 'electron';
 
 const keytarServiceName = "DiceKeys"
-IpcApiFactory.implementAsyncApi( "getDiceKey", (id: string) => keytar.getPassword(keytarServiceName, id));
-IpcApiFactory.implementAsyncApi( "setDiceKey", (id: string, humanReadableForm: string,) => keytar.setPassword(keytarServiceName, id, humanReadableForm));
-IpcApiFactory.implementAsyncApi( "deleteDiceKey", (id: string) => keytar.deletePassword(keytarServiceName, id));
-IpcApiFactory.implementAsyncApi( "getDiceKeys", async () => (await keytar.findCredentials(keytarServiceName))
-    .map(function (value: { account: string, password: string }) {
-        return {id: value.account, humanReadableForm: value.password}
+const keytarAccountName = "DiceKeysEncryptionKey"
+const storeFileName = "encrypted_dicekeys.json"
+
+type StoredDiceKeys = {
+    [id: string]: string;
+};
+
+interface EncryptionKey{
+    key: string,
+    iv: string
+}
+
+IpcApiFactory.implementAsyncApi( "setDiceKey", async (id: string, humanReadableForm: string) => {
+    const encryptionKey = await getEncryptionKey()
+    const cipher = crypto.createCipheriv("aes-256-gcm", encryptionKey.key, encryptionKey.iv)
+    const encryptedHumanReadableForm = cipher.update(humanReadableForm, "utf8", "base64")
+    let storedDiceKeys = await loadStoredDiceKeys()
+    storedDiceKeys[id] = encryptedHumanReadableForm
+    await saveStoredDiceKeys(storedDiceKeys)
+});
+
+
+IpcApiFactory.implementAsyncApi( "getDiceKey", getDiceKey);
+async function getDiceKey(id: string): Promise<string> {
+    const encryptionKey = await getEncryptionKey()
+    const decipher = crypto.createDecipheriv("aes-256-gcm", encryptionKey.key, encryptionKey.iv)
+
+    let storedDiceKeys = await loadStoredDiceKeys()
+    const encryptedHumanReadableForm = storedDiceKeys[id]
+
+    return new Promise<string>((resolve, reject) => {
+
+        if (encryptedHumanReadableForm !== null) {
+            resolve(decipher.update(encryptedHumanReadableForm, "base64", "utf8"))
+        } else {
+            reject("doesn't exists")
+        }
     })
-);
+
+}
+
+IpcApiFactory.implementAsyncApi( "getDiceKeys", async () => {
+    let storedDiceKeys = await loadStoredDiceKeys()
+
+    let unencrypted : {[id: string]: { id: string, humanReadableForm: string}} = {}
+    for (let id in storedDiceKeys) {
+        unencrypted[id] = {id: id, humanReadableForm: await getDiceKey(id)}
```

Comment:
Awaiting in a loop can be very time consuming.  You can instead spawn everything you want to await and then await them all.

```
const idDiceKeyPairs = await Promise.all( storedDiceKeyIds.map( async (id) => ([id, await getDiceKey(id)]) ) ):
// now use reduce to turn pairs into object
```